### DISCUSSION TITLE
Recommend Sending Leecher Message & Halting Uploads FIRST

### DISCUSSION THREAD

**Discussion by DeathStalker77** [nicotine-plus/nicotine-plus#2711] [Ideas]
Based on the Leecher plug-in settings, I would suggest sending the message FIRST and stopping the uploads until the Leecher complies with the user's settings.  Example:  I set the Min Folders to 25 & Min Files to 100 - the Leecher must have at least that amount of EACH before the downloads will be allowed to start.  The DEFAULT message would specify whatever the values the user has set.

This would prevent HnRs.  Also, FWIW, I am personally against the ability to have shares as PRIVATE (I don't know if that's been raised as a separate issue or not).  I have found that **most** users who have their shares as PRIVATE are usually selfish and demanding of specific asks of others in order for them to allow them access.  This goes against the very principle of file sharing.

Here is an example from one user (who I, reluctantly, will not dox - as I really believe that they should be exposed and publcly shamed) -
- Please only contact me if you have an item from my wishlist to exchange (see below) OR for 24bit / 96 khz releases, which i only have in 16bit 44khz.
OR if you can offer a CD Rip of CDs from the Magazines "Stereoplay", "AUDIO Magazin" or Record Label "In-Akustik" WHICH i DON'T HAVE ALREADY

And that is only one part of a lengthy list of demands.  IMHO, the file sharing community has no use for individuals like this.  If they want to ban the world, then let them do it one user at a time!

**Comment by slook** [comment]
It is problematic see issue https://github.com/nicotine-plus/nicotine-plus/issues/1565
